dbNSFP.pm

```diff
@@ -385,29 +442,38 @@ sub run {
     # make a clean copy as we're going to edit it
     %$data = %$tmp_data;
 
-    # convert data with multiple transcript values
-    # if($data->{Ensembl_transcriptid} =~ m/\;/) {
-
-    #   # find the "index" of this transcript
-    #   my @tr_ids = split(';', $data->{Ensembl_transcriptid});
-    #   my $tr_index;
-
-    #   for my $i(0..$#tr_ids) {
-    #     $tr_index = $i;
-    #     last if $tr_ids[$tr_index] =~ /^$tr_id(\.\d+)?$/;
-    #   }
-
-    #   next unless defined($tr_index);
-
-    #   # now alter other fields
-    #   foreach my $key(keys %$data) {
-    #     if($data->{$key} =~ m/\;/) {
-    #       my @split = split(';', $data->{$key});
-    #       die("ERROR: Transcript index out of range") if $tr_index > $#split;
-    #       $data->{$key} = $split[$tr_index];
-    #     } 
-    #   }
-    # }
+    # check and parse data if we're using transcript_match parameter
+    if($self->{transcript_match}) {
+
+      # find the "index" of this transcript
+      my @tr_ids = split(';', $data->{Ensembl_transcriptid});
+      my $tr_index;
+
+      for my $i(0..$#tr_ids) {
+        if($tr_ids[$i] eq $tr_id) {
+          $tr_index = $i;
+          last;
+        }
+      }
+
+      # if transcriptid doesn't match we have to nerf transcript-specific fields
+      if(!defined($tr_index)) {
+        for my $key(@{$self->{transcript_specific_fields}}) {
+          delete $data->{$key} if defined($data->{$key});
+        }
+      }
+
+      # refine values of transcript-specific fields
+      # we need to get the fields here as we're modifying $data in the loop
+      my @refine_fields = grep {defined($data->{$_}) && defined($self->{cols}->{$_})} @{$self->{transcript_specific_fields}};
+
+      foreach my $key(@refine_fields) {
+        my @split = split(';', $data->{$key});
+        die("ERROR: Transcript index out of range") if $tr_index > $#split;
```

Comment:
As @dglemos pointed out, instead of dying here, we can put a warning so that we can still get output from other fields that have valid values.

Reply:
Also, skip checking for fields that have '.' as value (i.e. - empty value)